### PR TITLE
NoUncheckedIndexedAccess strict checks

### PR DESCRIPTION
**Pull request by jonathanKingston**
**Asana Task/Github Issue:**

## Description

Enables `noUncheckedIndexedAccess` in `tsconfig.strict-core.json` to enforce stricter type checking for indexed access. This change required fixing 23 resulting type errors across 5 core files by adding appropriate type guards and assertions.

## Testing Steps

- Run `npm run tsc:strict-core` to confirm no new type errors.
- Run `npm run tsc` and `npm run lint` to ensure no regressions.

## Checklist

*Please tick all that apply:*

- [x] I have tested this change locally
- [ ] I have tested this change locally in all supported browsers
- [ ] This change will be visible to users
- [ ] I have added automated tests that cover this change
- [ ] I have ensured the change is gated by config
- [ ] This change was covered by a ship review
- [ ] This change was covered by a tech design
- [ ] Any dependent config has been merged

---
<p><a href="https://cursor.com/agents?id=bc-24eab4c5-9bc8-4b93-b5a2-cce358e5421e"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-24eab4c5-9bc8-4b93-b5a2-cce358e5421e"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</p>

---

> [!NOTE]
> **Low Risk**
> Mostly type-safety hardening and null-guard changes with minimal behavioral impact; the main runtime change is skipping/optionally logging missing feature constructors, which could hide misconfiguration if relied on.
> 
> **Overview**
> Enables TypeScript `noUncheckedIndexedAccess` in `tsconfig.strict-core.json`, tightening strict-core compilation rules for injected code.
> 
> To satisfy the new checks, this adds defensive guards/casts around indexed access across entry points and injected modules (e.g., optional feature lists, cookie parsing, canvas pixel indexing, URL parsing, and stack-trace parsing), and adds a debug-only warning when a bundled feature constructor is missing in `content-scope-features.js`.
> 
> Adds unit tests covering `computeEnabledFeatures`/`parseFeatureSettings` edge cases for platform-specific features and missing remote-config entries.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 94b7b9b7ef411b4fa9a1ac842aaccf032c4e9f88. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: injected/src/utils.js [hunk 1]
```diff
@@ -857,6 +862,7 @@ export function computeEnabledFeatures(data, topLevelHostname, platform, platfor
     const enabledFeatures = remoteFeatureNames
         .filter((featureName) => {
             const feature = data.features[featureName];
+            if (!feature) return false;
```
[the remark this answers is not in the document]
Comment (jonathanKingston) [reply injected/src/utils.js]: @cursoragent can you ensure that the always enabled features aren't dropped in this change?

File: injected/src/utils.js [hunk 2]
```diff
@@ -884,7 +890,9 @@ export function parseFeatureSettings(data, enabledFeatures) {
             return;
         }
 
-        featureSettings[featureName] = data.features[featureName].settings;
+        const feature = data.features[featureName];
+        if (!feature) return;
+        featureSettings[featureName] = feature.settings;
```
[the remark this answers is not in the document]
Comment (jonathanKingston) [reply injected/src/utils.js]: @cursoragent can you ensure that the always enabled features aren't dropped in this change?
